coordinator/src/services/translation/request/dosage.ts

```diff
@@ -581,17 +581,24 @@ function getListWithSeparators(list: Array<string>) {
   return elements
 }
 
-function stringifyRange(range: fhir.Range, pluralise = false) {
+function stringifyRange(range: fhir.Range, pluralise = false): Array<string> {
   const lowQuantity = range?.low
   const highQuantity = range?.high
   const lowUnit = stringifyQuantityUnit(lowQuantity, pluralise)
   const highUnit = stringifyQuantityUnit(highQuantity, pluralise)
-  const elements = [stringifyQuantityValue(lowQuantity)]
+  const lowValue = stringifyQuantityValue(lowQuantity)
+  const highValue = stringifyQuantityValue(highQuantity)
+
+  if (lowQuantity && !highQuantity) {
+    return ["at least ", lowValue, " ", lowUnit]
+  }
+  if (highQuantity && !lowQuantity) {
+    return ["up to ", highValue, " ", highUnit]
+  }
   if (lowUnit !== highUnit) {
-    elements.push(" ", lowUnit)
+    return [lowValue, " ", lowUnit, " to ", highValue, " ", highUnit]
   }
-  elements.push(" to ", stringifyQuantityValue(highQuantity), " ", highUnit)
-  return elements
+  return [lowValue, " to ", highValue, " ", highUnit]
```

Comment:
is there any reason we actually need to return this split array? could we not do something like 

`[${lowValue} to ${highValue} ${highUnit}]`

Reply:
just realised this wouldn't error if `lowValue/lowUnit/highValue/highUnit = null` so I think its better to leave as it